internal/headerbp/metrics.go

```diff
@@ -0,0 +1,72 @@
+package headerbp
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/reddit/baseplate.go/internal/prometheusbpint"
+)
+
+const (
+	clientMethodLabel = "client_method"
+	clientNameLabel   = "client_name"
+	headerNameLabel   = "header_name"
+	rpcTypeLabel      = "rpc_type"
+	serverMethodLabel = "server_method"
+	serviceLabel      = "service_name"
+)
+
+var (
+	clientHeadersRejectedTotal = promauto.With(prometheusbpint.GlobalRegistry).NewCounterVec(prometheus.CounterOpts{
+		Name: "baseplate_client_headers_rejected_total",
+		Help: "Total number of requests that were rejected by a client due to unapproved internal headers",
+	}, []string{
+		rpcTypeLabel,
+		serviceLabel,
+		clientNameLabel,
+		clientMethodLabel,
+		headerNameLabel,
+	})
+
+	clientHeadersSentTotal = promauto.With(prometheusbpint.GlobalRegistry).NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "baseplate_client_headers_sent_total",
+		Help:    "Total number of internal headers that were automatically sent by a client",
+		Buckets: []float64{1, 4, 8, 16, 32, 64, 128},
+	}, []string{
+		rpcTypeLabel,
+		serviceLabel,
+		clientNameLabel,
+		clientMethodLabel,
+	})
+
+	clientHeadersSentSize = promauto.With(prometheusbpint.GlobalRegistry).NewHistogramVec(prometheus.HistogramOpts{
+		Name:    "baseplate_client_headers_sent_size_total",
```

Comment:
`_sent_size_bytes`, `_total` is only for counters (this is a histogram), and there the unit is bytes.